### ISSUE TITLE
clippy::never_loop

### ISSUE THREAD

**Issue by i18nsite** [rust-lang/rust-clippy#13622]
### Description

![Image](https://github.com/user-attachments/assets/b20df982-d439-455f-a074-8f61e924ab03)

**Comment by samueltardieu** [comment]
As you point out the `#[allow(clippy::never_loop)]` lets you silent the lint for this particular case. I am not sure I understand what the issue is.

**Comment by Jarcho** [comment]
Blocks can have labels since 1.65. The lint should have an MSRV added for this case if it doesn't have one already, but it's not a high priority thing. (1.65 is two years old)

**Comment by i18nsite** [comment]
> As you point out the `#[allow(clippy::never_loop)]` lets you silent the lint for this particular case. I am not sure I understand what the issue is.

I wrote a state machine for parsing markdown, and used this technique in many places. It was annoying to add #![allow every time.

**Comment by samueltardieu** [comment]
As @Jarcho wrote, you can remove the outside `loop` and the latest `break`, and label your block instead:

```rust
'out: {
  …
  loop {
    …
    if … { continue; }
    …
    break 'out;
    …
  }
  …
}
```

If you insist on keeping the outer loop (which isn't a loop since you use `break`), you can also allow this Clippy lint globally in your library by adding the following fragment to your `Clippy.toml`:

```toml
[lints.clippy]
never_loop = "allow"
```

Would that work for you?

**Comment by i18nsite** [comment]
![Image](https://github.com/user-attachments/assets/5bc0f1db-a28f-4163-ba93-14c10664639d)
label block is still warn

**Comment by samueltardieu** [comment]
> label block is still warn

It warns because it is true: your loop never loops. It has no `continue` inside, and it breaks at the end of the loop so it does not iterate.

**Comment by nk9** [comment]
It seems like we can close this as there's no action to take on the Clippy side.

@rustbot  label S-triage
